docs/Webhook-Parameters.md

```diff
@@ -13,6 +13,8 @@ Usage of webhook:
         path to the json file containing defined hooks the webhook should serve, use multiple times to load from different files
   -hotreload
         watch hooks file for changes and reload them automatically
+  -http-methods string
+        globally restrict allowed HTTP methods; separate methods with comma
```

Comment:
Quick question, should this override per-hook settings? At the moment, we have the opposite behavior for headers, the per-hook setting overrides the global setting. With the current implementation, this isn't true for `http` methods because the request will be rejected by the `gorilla/mux` due to the `.Methods(...)`.

Reply:
Generally speaking, CLI options should always override config file options.  The issue here is that we're not overriding the per-hook option with a per-hook CLI option.  Their technically two separate items.

So, I could go either way here.  :smile:

One issue with the current PR is that the mux rejects the request without me catching it and logging a message.  If we moved the global check down into the hookHandler, we could do whatever we want (as to your question) and also log a similar message.

Reply:
I'm thinking we make the CLI option a "default allowed HTTP methods" where the a hook definition can override the default.  I'll work that up and add a new commit.

Reply:
Yes, that sounds about right to me :-)